### ISSUE TITLE
Cache last N messages per room to enable sed functionality globally.

### ISSUE THREAD

**Issue by Shmiddty** [Zirak/SO-ChatBot#232]
The title speaks for itself.

**Comment by CS1000** [comment]
Can't it just read last 100 messages in the room it was invoked and return the first one that can be replaced?  

Cache smells leaky...

**Comment by SomeKittens** [comment]
@CS1000 currently, it reads the DOM for the `!!s` command.  It can only read the DOM for the room that's currently up in the tab.

**Comment by CS1000** [comment]
what about reply-ing to a message like:  

    :2709976 s/search/replacement/gi

instead of `!!s`

**Comment by Shmiddty** [comment]
@CS1000 That's not a bad idea, but I don't think it solves the cross-room problem.

**Comment by CS1000** [comment]
@Shmiddty it would requre fetching the exact message id, abuse shouldn't be a problem with a timed execution. http://chat.stackoverflow.com/messages/22491127/history

**Comment by Zirak** [comment]
Replacing a specific message was handled in #220 though we still use the DOM. Honestly, anything to do with messages that isn't in the current DOM is blocked on the same thing #27 (and many more things) are blocked on: Specifically how the bot has no internal representation of the room it's in or the chat in general.

**Comment by Zirak** [comment]
owait, this is a duplicate of #216.
